src/compiler/optim.jl

```diff
@@ -51,9 +51,18 @@ function optimize!(ctx::CompilerContext, mod::LLVM.Module, entry::LLVM.Function)
 
         ModulePassManager() do pm
             initialize!(pm)
+
+            # lower intrinsics
             add!(pm, FunctionPass("LowerGCFrame", lower_gc_frame!))
             aggressive_dce!(pm) # remove dead uses of ptls
             add!(pm, ModulePass("LowerPTLS", lower_ptls!))
+
+            # the Julia GC lowering pass also has some clean-up that is required
+            function LLVMAddLateLowerGCFramePass(PM::LLVM.API.LLVMPassManagerRef)
+                LLVM.@apicall(:LLVMExtraAddLateLowerGCFramePass,Cvoid,(LLVM.API.LLVMPassManagerRef,), PM)
```

Comment:
Is there a PR to Base for this already? @yingboma and I were working on exposing all the Julia passes + barrier the other day but got sidetracked.

Reply:
Nicely spotted. No, I just added this. Turns out we need the pass its `CleanupIR` functionality to get rid of operation bundles; it doesn't do anything otherwise. Should I put this up, or are you going to upstream your work?

Reply:
I was running into some linker issues, so maybe put your's up and I will add the other passes.

Reply:
https://github.com/JuliaLang/julia/pull/31428